src/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -173,16 +168,28 @@ private Type GetEquivalentType(Type type)
                 }
                 if (equiv == null)
                 {
-                    var interfacesWithInfo = info.ImplementedInterfaces.Select(i => new { Type = i, Info = i.GetTypeInfo() }).ToArray();
-                    var singleTypeGenInterfacesWithGetType = interfacesWithInfo
-                        .Where(i => i.Info.IsGenericType && i.Info.GenericTypeArguments.Length == 1)
-                        .Select(i => new { Type = i.Type, Info = i.Info, GenType = i.Info.GetGenericTypeDefinition() });
-                    Type typeArg = singleTypeGenInterfacesWithGetType.Where(i => i.GenType == typeof(IOrderedQueryable<>) || i.GenType == typeof(IOrderedEnumerable<>)).Select(i => i.Info.GenericTypeArguments[0]).Distinct().SingleOrDefault();
+                    var interfacesWithInfo =
+                        info.ImplementedInterfaces.Select(IntrospectionExtensions.GetTypeInfo).ToArray();
+                    var singleTypeGenInterfacesWithGetType =
+                        interfacesWithInfo.Where(i => i.IsGenericType && i.GenericTypeArguments.Length == 1)
+                            .Select(i => new {Info = i, GenType = i.GetGenericTypeDefinition()})
+                            .ToList();
+                    Type typeArg =
+                        singleTypeGenInterfacesWithGetType.Where(
+                            i => i.GenType == typeof(IOrderedQueryable<>) || i.GenType == typeof(IOrderedEnumerable<>))
```

Comment:
Nit: I'd personally prefer to outline all the `.Where`, `.Select`, etc. underneath each other in one column rather than having the `.Where` sitting on the first line. Same higher up and down below.